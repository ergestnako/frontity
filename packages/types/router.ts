import Package from "./package";

interface Router extends Package {
  settings: {
    router: {};
  };
}

export default Router;
