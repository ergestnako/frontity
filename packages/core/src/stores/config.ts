import { Namespace } from "@frontity/types/namespace";
import { namespaced } from "overmind/config";
import { Settings } from "./settings/types";

export default ({
  namespaces,
  settings
}: {
  namespaces: { [key: string]: Namespace };
  settings: Settings;
}) =>
  namespaced(
    Object.entries(namespaces).reduce(
      (config, [namespace, module]) => {
        if (module.Store) config[namespace] = module.Store;
        return config;
      },
      { settings }
    )
  );
