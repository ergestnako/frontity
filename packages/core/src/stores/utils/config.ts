import { Namespace } from "@frontity/types/namespace";
import { namespaced } from "overmind/config";

export default ({ namespaces }: { namespaces: { [key: string]: Namespace } }) =>
  namespaced(
    Object.entries(namespaces).reduce((config, [namespace, module]) => {
      if (module.Store) config[namespace] = module.Store;
      return config;
    }, {})
  );
