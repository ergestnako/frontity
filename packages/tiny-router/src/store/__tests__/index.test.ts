import { createOvermindMock } from "overmind";
import { namespaced } from "overmind/config";
import router from "..";

const settings = {
  state: {
    frontity: {
      url: "https://example.com"
    }
  }
};

let overmind = createOvermindMock(
  namespaced({
    settings,
    router
  })
);

afterEach(() => {
  overmind = createOvermindMock(
    namespaced({
      settings,
      router
    })
  );
});

describe("state", () => {
  test("url", () => {
    expect(overmind.state.router.url).toBe("https://example.com/");
  });
});

describe("actions", () => {
  test("set", async () => {
    const mutations = await overmind.actions.router.set("/some-post");
    expect(mutations).toMatchSnapshot();
  });
});
