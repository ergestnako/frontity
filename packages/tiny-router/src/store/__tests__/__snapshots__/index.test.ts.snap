// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions set 1`] = `
Array [
  Object {
    "args": Array [
      "/some-post",
    ],
    "method": "set",
    "path": "path",
  },
  Object {
    "args": Array [
      null,
    ],
    "method": "set",
    "path": "page",
  },
]
`;
